### DISCUSSION TITLE
routes 数组是如何被动态加载的

### DISCUSSION THREAD

**Discussion by arlyxiao** [umijs/umi#8858] [General]
我尝试在程序运行时主动 import routes 数组定义的 component，会提示 can't find module

想了解下 umi 这边大概是怎么做的？

实现代码主要在哪部分？

**Comment by fz6m** [comment]
由于缺乏足够的信息，请提供 [最小重现](https://stackoverflow.com/help/minimal-reproducible-example) 的 demo 帮助排查问题。

**Reply by arlyxiao** [reply]
这种方式的写法会出现 can't find module `@/pages/Welcome`
```   
const name = "@/pages/Welcome"
    import(name).then((mod) => {
      console.log(mod.default)
    })
```

如果换成这样，就没问题
```   
    import("@/pages/Welcome").then((mod) => {
      console.log(mod.default)
    })
```

也就是说，在运行时，无法直接 import 变量，
如果想在运行时，import 变量，这里的变量是指 routes 数组定义的 component 路径, 有什么办法嘛？

**Reply by fz6m** [reply]
给一个 demo 仓库看看
